elevation_layer/src/elevation_layer.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * elevation_layer.cpp
+ *
+ *  Created on: Nov 5, 2018
+ *      Author: Eugenio Chisari
+ *	 Institute: ANYbotics
+ */
+
+#include <elevation_layer/elevation_layer.h>
+#include <pluginlib/class_list_macros.h>
+
+PLUGINLIB_EXPORT_CLASS(elevation_layer::ElevationLayer, costmap_2d::Layer)
+
+using costmap_2d::FREE_SPACE;
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::NO_INFORMATION;
+
+using costmap_2d::Observation;
+using costmap_2d::ObservationBuffer;
+
+namespace elevation_layer {
+
+ElevationLayer::ElevationLayer()
+    : filterChain_("grid_map::GridMap"), elevation_map_received_(false), filters_configuration_loaded_(false) {
+  costmap_ = nullptr;  // this is the unsigned char* member of parent class Costmap2D.
+}
+
+void ElevationLayer::onInitialize() {
+  ros::NodeHandle nh("~/" + name_);
```

Comment:
Sorry, I thought I added this comment already, but I cannot find it anymore. Could you get the node handle from your parent class instead of creating 3 new node handles in your methods?

Reply:
How? I don't understand where do I have 3 nodehandles, and also i am not sure how to get from the parent  class. Which class do you mean?

Reply:
Ok Hamish showed me the  3 nodehandles. I am still not sure how to get from parent class. What about i do one nodehandle as class member, initialize it in constructor, and use it 
everywhere?

Reply:
If your parent class costmap_2d::CostmapLayer is using ROS, it most probably has already a node handle, which you could access and use. If not, your proposition is the way to go :)